theano/tensor/basic.py

```diff
@@ -2588,6 +2588,175 @@ def nonzero_values(a):
     return a.flatten()[flatnonzero(a)]
 
 
+class UnravelIndex(gof.Op):
+    __props__ = ('order',)
+
+    def __init__(self, order='C'):
+        self.order = order
+
+    def make_node(self, indices, dims, ndim):
```

Comment:
This isn't super. We expect at some places in Theano that we can re-create a new node just from the inputs. To respect that, you would need to also add ndim to the inputs of this node just to be sure to don't have problems elsewhere. As it will be a constant, it should not be a problem.

I would do ndim=None and call get_vector_lenght() here if ndim is None.

Reply:
I moved `ndim` to `__init__` instead, because that seemed the easiest solution. It's now similar to how it is done in `Reshape`.